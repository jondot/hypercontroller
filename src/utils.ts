import { replace, kebabCase } from 'lodash'

const makePath = (name: string) => {
  if (name === 'index') {
    return ''
  }
  const controllerOrPath = replace(name, /Controller$/, '')
  if (controllerOrPath === 'Index') {
    return ''
  }

  return kebabCase(controllerOrPath)
}
const leadingSlash = (mountedPath: string) =>
  mountedPath.startsWith('/') ? mountedPath : `/${mountedPath}`

export { makePath, leadingSlash }
