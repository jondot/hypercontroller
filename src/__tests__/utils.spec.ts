import { makePath } from '../utils'

describe('utils', () => {
  it('should convert to HTTP paths', () => {
    expect(makePath('FooController')).toMatchSnapshot('controller')
    expect(makePath('index')).toMatchSnapshot('index')
    expect(makePath('FooBarController')).toMatchSnapshot('kebab')
    expect(makePath('indexFoobar')).toMatchSnapshot('non-index')
    expect(makePath('hello')).toMatchSnapshot('normal')
  })
})
