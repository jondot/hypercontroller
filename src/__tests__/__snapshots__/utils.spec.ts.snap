// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils should convert to HTTP paths: controller 1`] = `"foo"`;

exports[`utils should convert to HTTP paths: index 1`] = `""`;

exports[`utils should convert to HTTP paths: kebab 1`] = `"foo-bar"`;

exports[`utils should convert to HTTP paths: non-index 1`] = `"index-foobar"`;

exports[`utils should convert to HTTP paths: normal 1`] = `"hello"`;
